### ISSUE TITLE
List of Signatures

### ISSUE THREAD

**Issue by loreberti89** [ministero-salute/dcc-utils#7]
Hello, my issue is a question.

In the example verify_signature_from_list.js I have seen that is used this url
`https://raw.githubusercontent.com/bcsongor/covid-pass-verifier/35336fd3c0ff969b5b4784d7763c64ead6305615/src/data/certificates.json`

for get the certificates, but I can use this list and it is always valid or it is only for example purpose?

In this case, Is there a url where we can get the valid certificates?

Thankyou.

**Comment by rebuglio** [comment]
Swedish government make a public API: https://dgcg.covidbevis.se/tp/ listing all europe certificates
It would be useful if Italy did the same. 

Swedish api can be used with dcc-utils with some adjust, es:

```
const fetch = require('node-fetch');
const DCC = require('../src/dcc');
const jwtdecode = require('jwt-decode');
const {webcrypto, verify, cbor} = require("cosette/build/sign");

(async () => {
  const response = await fetch('https://dgcg.covidbevis.se/tp/trust-list');
  const jws = jwtdecode(await response.text());

  const jwtkeys = Object
    .values(jws.dsc_trust_list)
    .flatMap((country) => country.keys);

  const dcc = await DCC.fromRaw('HC1:...');

  // verify, throw exception if signature mismatch
  const keyalg = { "hash": { "name": "SHA-256" }, "name": "ECDSA", "namedCurve": "P-256" };
  const cbor_payload = await verify(dcc._coseRaw, async (kid) => {
    const key = jwtkeys.find(k => k.kid === kid.toString('base64'));
    return {key: await webcrypto.subtle.importKey(
      'jwk', key, keyalg, true, ['verify'])}
  });

  const payload = dcc.payload()
})();
```

Nb: this code isn't optimized, since dcc-utils and cosette.verify do the same cbor decoding one time

**Comment by loreberti89** [comment]
> I don't know about bcsongor list, he is a private individual who has created a list, I do not think he can be considered 100% reliable. Lovasoa make a similar list: https://raw.githubusercontent.com/lovasoa/sanipasse/master/src/assets/Digital_Green_Certificate_Signing_Keys.json
> 
> Swedish government make a public API: https://dgcg.covidbevis.se/tp/ listing all europe certificates
> It would be useful if Italy did the same.
> 
> Swedish api can be used with dcc-utils with some adjust, es:
> 
> ```
> const fetch = require('node-fetch');
> const DCC = require('../src/dcc');
> const jwtdecode = require('jwt-decode');
> const {webcrypto, verify, cbor} = require("cosette/build/sign");
> 
> (async () => {
>   const response = await fetch('https://dgcg.covidbevis.se/tp/trust-list');
>   const jws = jwtdecode(await response.text());
> 
>   const jwtkeys = Object
>     .values(jws.dsc_trust_list)
>     .flatMap((country) => country.keys);
> 
>   const dcc = await DCC.fromRaw('HC1:...');
> 
>   // verify, throw exception if signature mismatch
>   const keyalg = { "hash": { "name": "SHA-256" }, "name": "ECDSA", "namedCurve": "P-256" };
>   const cbor_payload = await verify(dcc._coseRaw, async (kid) => {
>     const key = jwtkeys.find(k => k.kid === kid.toString('base64'));
>     return {key: await webcrypto.subtle.importKey(
>       'jwk', key, keyalg, true, ['verify'])}
>   });
> 
>   const payload = dcc.payload()
> })();
> ```
> 
> Nb: this code isn't optimized, since dcc-utils and cosette.verify do the same cbor decoding one time

Yes, I think is not 100% reliable.
But I don't understand why use a indivudial list on official repo instead of a official list certificates.

Anyway, on VerificaC-19 App, we can found 2 url.
https://get.dgc.gov.it/v1/dgc/signercertificate/update
https://get.dgc.gov.it/v1/dgc/signercertificate/status
So I think these are 2 reliable link, but I dont' undestand yet, how this data is used for validation.

So Yes,  It would be useful if Italy provided a "api" for get certificates and then use this library for validate green pass.
But, my question in this case would be:

Is it legal to verify a green pass with an different app than Verification C-19 but that uses the libraries of the Ministero della salute? Maybe not, so could be the reason why the "api" are not yet present.

**Comment by astagi** [comment]
@loreberti89 yes, that list is only for example purpose! Dcc-utils library is intended to be used only for testing/debugging/study purpose, it's just a collection of utilities. [Here](https://github.com/ministero-salute/dcc-utils/issues/1) you can see how to use the API you mentioned.

**Comment by loreberti89** [comment]
> @loreberti89 yes, that list is only for example purpose! Dcc-utils library is intended to be used only for testing/debugging/study purpose, it's just a collection of utilities. [Here](https://github.com/ministero-salute/dcc-utils/issues/1) you can see how to use the API you mentioned.

Thankyou for your reply.
Ok, so now I can use the official API and then validate GP.
But my question now is:

Is it legal to verify a green pass with an different app than Verification C-19 but that uses the libraries of the Ministero della salute?

Example, I have created an app for Access Control in workplace, and then I would use GreenPass for access, but actually the validity of GreenPass is valid only with Verifica C-19 ?
If I use this library that Valid the GreenPass, the owner of company will also need to verify validity of GP thourgh Verifica C-19?

**Comment by valeprog** [comment]
Hi I'm able to decode Grenne pass and verify it with this link:
https://get.dgc.gov.it/v1/dgc/signercertificate/update

But how can I use this one?
https://get.dgc.gov.it/v1/dgc/signercertificate/status
Thanks!

**Comment by astagi** [comment]
Hi @valeprog that endpoint returns a list of kids, you need to use `/update` endpoint to get the entire certificate

**Comment by valeprog** [comment]
Ok thanks, so I can use only `/update` endpoint correct?

**Comment by astagi** [comment]
Yes @valeprog! [Here](https://github.com/ministero-salute/dcc-utils/issues/1) you can see how to use `/update` endpoint correctly.

**Comment by valeprog** [comment]
Good thanks!

**Comment by rene-vertical-life** [comment]
@astagi any news on the official SDK?

**Comment by astagi** [comment]
@rene-vertical-life nope
